mineral-ui/mineral-ui
# [Theme] [Website] [All] Rework spacing variables

### Description


> Note for reviewers: You'll want to review this commit-by-commit, for the reasons detailed below.

#### Commit 1
- Add `space_inline_xx`, `space_inset_xx`, and `space_stack_xx` theme variables, per [spec](https://ca.invisionapp.com/d/main#/console/11070131/250301138/preview)
- Remove `spacing_xx` theme variables
- Update components to use the new variables
- Remove CardRow and cardTheme, which should have been done with #307 
- Add an exported `componentTheme` to MenuGroup and reference appropriately

#### Commit 2
- Update website components, including anything in `/demos/`, to use the new variables
- Refactor LiveEditor maxHeight
- Various small style improvements

#### Commit 3
- Update snapshots after modifying theme
    - Every component snapshot should have changed because the theme changed, but the _styles_ should not change at all
        - The kinda-exception to this is the Custom Render example, which has some styles in the source that reference the theme

#### Commit 4
- While updating and reviewing snapshots, I realized that the Menu spec was testing the snapshots for not just Menu, but all of the Menu* components. This corrects that. It's a separate commit because it makes the snapshots really hard to review for meaningful changes in the previous commit.

### Motivation and context

Closes #340 

### Screenshots, videos, or demo, if appropriate


https://340-spacing-tokens--mineral-ui.netlify.com/

### How to test


1. `cd mineral-ui && npm start`
2. Check all of the library components for proper appearance, comparing against http://mineral-ui.com
3. Check the site components for proper appearance, comparing against http://mineral-ui.com
4. Confirm that the snapshots indicate no style changes, only theme changes

### Types of changes

- Breaking change (fix or feature that would cause existing functionality to change)

Removes some theme variables.

### Checklist




### How does this PR make you feel?

![Space](https://media.giphy.com/media/TZf4ZyXb0lXXi/giphy.gif)
